src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -109,11 +97,16 @@ public void InvokeCompletionCallback(uint sniErrorCode)
         /// <summary>
         /// Allocate space for data
         /// </summary>
-        /// <param name="capacity">Bytes to allocate</param>
-        public void Allocate(int capacity)
+        /// <param name="bufferSize">Length of byte array to be allocated</param>
+        public void Allocate(int bufferSize)
         {
-            _capacity = capacity;
-            _data = new Byte[capacity];
+            if (_data == null || _data.Length != bufferSize)
```

Comment:
I asked this last time, but I don't remember the answer.  Why can't this be `_data.Length < bufferSize`?

Reply:
There were 2 reasons:
1. I changed to `_data.Length < bufferSize`, and faced several test failures during network operation. It seemed they needed exact packet length as a parameter the tests handed over to `Allocate()` method. So I rolled back to original implementation.
2. The packet size is initially 4,096 for login process, and then becomes 8,000 constantly until connection is closed. So, in this situation, there is not much benefit for changing to `_data.Length < bufferSize`.